yt_dlp/extractor/afreecatv.py

```diff
@@ -347,77 +347,56 @@ class AfreecaTVLiveIE(AfreecaTVIE):  # XXX: Do not subclass from concrete IE
-            aid = traverse_obj(aid_response, ('CHANNEL', 'AID'))
-            if not aid:
-                continue
-
-            stream_base_url = channel_info.get('RMD') or 'https://livestream-manager.afreecatv.com'
-            stream_info = self._download_json(
-                f'{stream_base_url}/broad_stream_assign.html', broadcast_no, fatal=False,
-                query={
-                    'return_type': channel_info.get('CDN', 'gcp_cdn'),
-                    'broad_key': f'{broadcast_no}-common-{quality_str}-hls',
-                },
-                note=f'Downloading metadata for {quality_str} stream',
-                errnote=f'Unable to download metadata for {quality_str} stream') or {}
-
-            if stream_info.get('view_url'):
-                formats.append({
-                    'format_id': quality_str,
-                    'url': update_url_query(stream_info['view_url'], {'aid': aid}),
-                    'ext': 'mp4',
-                    'protocol': 'm3u8',
-                    'quality': quality_key(quality_str),
-                })
-
-        station_info = self._download_json(
-            'https://st.afreecatv.com/api/get_station_status.php', broadcast_no,
-            query={'szBjId': broadcaster_id}, fatal=False,
-            note='Downloading channel metadata', errnote='Unable to download channel metadata') or {}
+        params = {'bno': broadcast_no, 'stream_type': 'common', 'type': 'aid', 'quality': 'master', 'pwd': password}
+
+        aid_response = self._download_json(self._LIVE_API_URL, broadcast_no, data=urlencode_postdata(params),
+                                           note='Downloading access token for stream',
+                                           errnote='Unable to download access token for stream')
+        aid = traverse_obj(aid_response, ('CHANNEL', 'AID'))
+        if not aid:
+            raise ExtractorError(f'Unable to extract AID from {aid_response}', expected=False)
+
+        stream_base_url = channel_info.get('RMD') or 'https://livestream-manager.afreecatv.com'
+        stream_info = self._download_json(f'{stream_base_url}/broad_stream_assign.html', broadcast_no, query={
+            # works: gs_cdn_pc_app, gs_cdn_mobile_web, gs_cdn_pc_web
+            'return_type': 'gs_cdn_pc_app',
+            'broad_key': f'{broadcast_no}-common-master-hls',
+        }, note='Downloading metadata for stream', errnote='Unable to download metadata for stream')
+
+        station_info = self._download_json('https://st.afreecatv.com/api/get_station_status.php', broadcast_no,
+                                           query={'szBjId': broadcaster_id}, note='Downloading channel metadata',
+                                           errnote='Unable to download channel metadata')
+
+        formats = self._extract_m3u8_formats(update_url_query(stream_info['view_url'], {'aid': aid}),
+                                             broadcast_no, 'mp4', m3u8_id='hls', headers={'Referer': url})
```

Comment:
let's flip-flop these two requests, since `formats` is fatal and `station_info` shouldn't be (it's metadata-only)
```suggestion
        formats = self._extract_m3u8_formats(
            stream_info['view_url'], broadcast_no, 'mp4', m3u8_id='hls',
            query={'aid': aid}, headers={'Referer': url})

        station_info = traverse_obj(self._download_json(
            'https://st.afreecatv.com/api/get_station_status.php', broadcast_no,
            'Downloading channel metadata', 'Unable to download channel metadata',
            query={'szBjId': broadcaster_id}, fatal=False), {dict}) or {}
```